lib/que/locker.rb

```diff
@@ -512,5 +519,19 @@ def mark_id_as_locked(id)
         "Tried to lock a job that was already locked: #{id}"
       end
     end
+
+    def write_pid
+      return unless @pidfile
+
+      File.open(@pidfile, ::File::CREAT | ::File::EXCL | ::File::WRONLY) do |f|
```

Comment:
I haven't seen these options given to file opening before - what are they doing? / why are they necessary here? To cater for if the file already exists? If it's important, maybe it's worth adding a test?

Reply:
You right. Change a little :)